### ISSUE TITLE
Issue with implicit cast between different base scalar types

### ISSUE THREAD

**Issue by vpinna80** [sdmx-twg/vtl#383]
Hi, with the introduction of implicit casts for any conversion without mask, a problem arises.
Consider the following example:
```
ds[filter between(value, 10, 100)]
```
where `value` is of type string. 

Which cast should be performed in this case? The new specification does not say.
Clearly casting string to integer could cause problems with the output, whereas casting integer to string could cause problems with sorting (i.e. "9" > "100").
I think that explicit cast should be reintroduced when converting string to numbers (or perhaps the opposite).

**Comment by antonio-olleros** [comment]
I don't undertand your comment well. And why and how do you know that OBS_VALUE is number?

**Comment by vpinna80** [comment]
From the definition of the variable in the structure



[https://www.bancaditalia.it/application/themes/bdi/images/header/marchio-BI-stampa-80.png]
VALENTINO PINNA
Dipartimento di Economia e Statistica
Supporto Informatico Dip. ECS
Invia un messaggio o Chiama su ***@***.***>
Palazzo Koch, via Nazionale 91
00184 - Roma

**Comment by antonio-olleros** [comment]
Well, but this does not need to be defined!!! It is something we are calculating, we are always deriving the result, not asking the users to know the result of the structure

**Comment by NicoLaval** [comment]
Implicit casting of integer to number is natural.

For all other cases, I prefer to oblige users to explicitly express castings, to make the script easier to understand and avoid "magic" effects that are difficult to maintain for a new contributor.

**Comment by antonio-olleros** [comment]
I agree. I think that the current implicit casting works well and makes sense, I would keep it as it is

**Comment by linardian** [comment]
Merging this thread with the "dual" one (thousand separators) we could agree that from string to number/integer eand viceversa we should go towards adopting explicit cast. If you do agree, we will follow this direction.
